framework/xds_k8s_testcase.py

```diff
@@ -411,6 +411,55 @@ def assertSuccessfulRpcs(
         )
         return lb_stats
 
+    # Return number of servers using subchannels
+    def getActiveNumberOfServers(
+        self,
+        test_client: XdsTestClient,
+        secure_channel: bool = False,
+    ) -> int:
+        channel = test_client.wait_for_server_channel_state(
+            state=_ChannelState.READY
+        )
+        subchannels = list(
+            test_client.channelz.list_channel_subchannels(channel)
+        )
+        return len(subchannels)
+
+    def _assertActiveNumberOfServers(
+        self,
+        expected_num_servers: int,
+        test_client: XdsTestClient,
+        secure_channel: bool = False
+    ) -> None:
+        self.assertEqual(
+            self.getActiveNumberOfServers(test_client, secure_channel),
+            expected_num_servers)
+
+    def assertActiveNumberOfServers(
+        self,
+        expected_num_servers: int,
+        test_client: XdsTestClient,
+        secure_channel: bool = False,
+        retry_timeout: dt.timedelta = TD_CONFIG_MAX_WAIT,
+        retry_wait: dt.timedelta = dt.timedelta(seconds=30)  
+    ) -> None:
+        retryer = retryers.constant_retryer(
+            wait_fixed=retry_wait,
+            timeout=retry_timeout,
+            log_level=logging.INFO,
+            error_note=(
+                f"Didn't find the expected number of active servers {expected_num_servers}"
+                f" before timeout {retry_timeout} (h:mm:ss)"
+            ),
+        )
+        retryer(
+            self._assertActiveNumberOfServers,
+            expected_num_servers,
+            test_client,
+            secure_channel,
+        )
```

Comment:
FYI - In cases like this it's more convenient to use retryer's context manager feature instead of defining a separate method just for a single assert. Example: https://github.com/grpc/psm-interop/blob/b57b402246cee51058cdb4c41dd80d72ab0050e5/framework/xds_k8s_testcase.py#L711-L718